### PR TITLE
Move to webpack 5

### PR DESCRIPTION
**Pull request by martinpitt**
Mostly cherry-picked from recent starter-kit PRs. 

 - [x] prerequisite commits: PR #704
 - [x] Fix scrolling to "used by" container to not crash for stopped containers (regression in f1e7c9b1518dbb): PR #705
 - [x] Get rid of react-scrollable-anchor, which is dead and does not work any more with webpack 5: PR #707 
 - [x] fix terser-built LICENSE file

### CODE REVIEW

**Comment by martinpitt** [comment]
Oops, the automatic dist/index.js.LICENSE.txt isn't being built any more.

**Comment by martinpitt** [comment]
Hah, so we do need terser-webpack-plugin for the LICENSE bits still. I'll send a PR to starter-kit to enable that, as we had to do that in -podman and -machines.

**Comment by martinpitt** [comment]
This crashes in node_modules/react-scrollable-anchor/lib/utils/func.js, which is shipped (by its own npm module) as:
```js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _arguments = arguments;
var debounce = exports.debounce = function debounce(func, wait, immediate) {
[...]
```
That `arguments` is not in the [original upstream source](https://github.com/gabergg/react-scrollable-anchor/blob/master/src/utils/func.js), so I suppose some babeling in there somehow broke that.

The same code is present in the webpack in master, but apparently embedded differently. Either way, [react-scrollable-anchor](https://www.npmjs.com/package/react-scrollable-anchor) has not been updated in 3 years, so it's probably time to replace that with something current PF4-ish.

File: src/Containers.jsx [hunk 1]
```diff
@@ -226,12 +225,12 @@ class Containers extends React.Component {
         }
 
         return {
-            expandedContent: <ScrollableAnchor id={container.Id} key={container.Id}>
```
Comment (martinpitt) [inline src/Containers.jsx:229]: Wait, I misunderstood what this does -- it was introduced in commit b5f8df880253e2d. I'll check what that actually renders to, and whether it's enough to just replace it with a simple HTML tag.

Comment (martinpitt) [reply src/Containers.jsx:229]: Lol -- this functionality is broken in master. Once I click on the "used by" container, the page crashes with `TypeError: props.showAll is not a function`.

Comment (martinpitt) [reply src/Containers.jsx:229]: Fixed in #705, but the actual scrolling is still broken.

**Comment by martinpitt** [comment]
All dependencies are fixed, so this PR is now a rather simple change. Tested locally (also visually) in dev and prod modes.

**Comment by martinpitt** [comment]
Meh -- the internet is full of questions how to *disable* creation of a LICENSE.txt file, but I can't get terser to *build* it now, even though it should.. again needswork.

**Comment by martinpitt** [comment]
Ugh, seems that webpack 5 uses a bit more memory which pushes packit over the line. I can reproduce this locally with
```
git clean -ffdx; podman run -ti -e NODE_OPTIONS=--max-old-space-size=500 --rm --memory 768MB -v $PWD:/src -w /src docker.io/usercont/sandcastle:prod make dist-gzip
```
I experimented with other `--max-old-space-size` values, but there is no sweet spot any more. As webpack is not at all security relevant, I am not willing to pay the price of setting up pre-built -dist for podman right now. Let's just keep webpack 4 for the time being.

At least this wasn't in vain, the recent fixes were all worth the effort :grin:

**Comment by martinpitt** [comment]
Found a workaround -- building in development mode is fine.

**Comment by martinpitt** [comment]
All green! :four_leaf_clover:

**Review by marusak — APPROVED** [review]
Danke!

**Pull request merged**